lib/Common/Memory/PageAllocator.h

```diff
@@ -479,24 +479,131 @@ class PageAllocatorBase: public PageAllocatorBaseCommon
     static size_t GetAndResetMaxUsedBytes();
 
 #if ENABLE_BACKGROUND_PAGE_FREEING
+    struct FreePageEntry
```

Comment:
Can you add a TODO item to implement a platform agnostic version of interlocked linked lists?